src/twisted/protocols/ftp.py

```diff
@@ -667,7 +667,7 @@ def __init__(self, cons):
             self.write = cons.write
 
     def write(self, bytes):
-        return self.cons.write(bytes.replace(os.linesep, "\r\n"))
+        return self.cons.write(bytes.replace(os.linesep.encode(), b"\r\n"))
```

Comment:
Probably worth setting
```
_LINESEP_BYTES = os.linesep.encode()
```
somewhere

Reply:
Done at the top of the file, along with _CRLF_BYTES constant.